### ISSUE TITLE
"!!MISSING: Command" after upgrade to 5.9.0

### ISSUE THREAD

**Issue by pkese** [ionide/ionide-vscode-fsharp#1620]
After upgrade to 5.9.0 I'm seeing `!!MISSING: Command` in code lens for every line.

Downgrading Ionide back to 5.8.1 fixed the issue.

This is a ssh-remote seession from Linux to Linux, both with dotnet 5.0.402.

**Comment by isaacabraham** [comment]
Seeing this, too.

**Comment by ed-ilyin** [comment]
Same on MacOS and dotnet 6

**Comment by beauvankirk** [comment]
If anyone is able to verify this issue is resolved by [the PR I submitted here](https://github.com/ionide/ionide-vscode-fsharp/pull/1626), I can open the 'real' one on FSAC as well/instead.  (More info in the PR)

**Comment by baronfel** [comment]
Closing, as this was implemented by @beauvankirk and released as part of 5.9.1.

**Comment by pkese** [comment]
@beauvankirk @baronfel 

I'm still getting this with Ionide 5.9.1.
But what is really interesting is that after a while it starts working for new files that I open.

It is as if the FSAC takes a while to start responding to Ionides' requests (*that's just my hypothesis, there may be other causes*) and everything that Ionide analyzes up to that point is presented as "!!MISSING: Command!! | !!MISSING: Command!!".

But files that I open later get proper code lens hints.

Problem is that Ionide caches those initial 'missing' responses and won't update.

**Comment by pkese** [comment]
And sometimes it resolves some lines completely and some just partially:
- both references and type on the top
- only type on the bottom
![image](https://user-images.githubusercontent.com/278492/141806274-51cfd7a6-0897-44f8-a444-2630476ab375.png)

**Comment by pkese** [comment]
Unfortunately I can't reopen the ticket.

**Comment by pkese** [comment]
No. My hypothesis was wrong. It's not that after a while FSAC starts working... rather it stays inconsistent.  Sometimes it works, then after a while I open another file and it doesn't.

**Comment by realparadyne** [comment]
I am seeing this on multiple Windows PC's using 6.0.6.  But it's working ok editing a project on WSL.

<img width="247" alt="image" src="https://user-images.githubusercontent.com/2536830/176514744-5fe2a0f1-04c7-450c-b233-c8514c92edc3.png">

**Comment by psfinaki** [comment]
Can reproduce it in 7.3.0:
![image](https://user-images.githubusercontent.com/5451366/195353125-7352176a-0992-485e-924d-4e1e39248ab6.png)
Switching to LineLens doesn't help - the info just disappears altogether.

Any workaround for this?

**Comment by pkese** [comment]
@psfinaki I've found out that sometimes, I'd just close that editor (file within the project) and open it again and it would display normally ... as if it's just initial load that fails to work.
Is that your experience as well?

**Comment by psfinaki** [comment]
@pkese well yes, just what you say - _sometimes_ it helps :) which is, well, annoying and mysterious.

**Comment by realparadyne** [comment]
I am still getting this or other similar issues across multiple PC's.  I keep wanting to make use of VS Code + Ionide for projects but sadly it seems that 80+% of the time (across multiple systems) it's just non-functional so I must stick with full VS 😒

I just tried various combinations of closing and opening, closing restarting, and then opening the editor on this new PC with recent fresh installs of everything, and it's just the MISSING command and no intellisense.

Is there some procedure for capturing detailed diagnostics that will pinpoint what is configured wrongly on this device?  What do I have installed that's the wrong version?  What did I install in the wrong order?

**Comment by realparadyne** [comment]
I am still getting this or other similar issues across multiple PC's.  I keep wanting to make use of VS Code + Ionide for projects but sadly it seems that 80+% of the time (across multiple systems) it's just non-functional so I must stick with full VS 😒

I just tried various combinations of closing and opening, closing restarting, and then opening the editor on this new PC with recent fresh installs of everything, and it's just the MISSING command and no intellisense.

Is there some procedure for capturing detailed diagnostics that will pinpoint what is configured wrongly on this device?  What do I have installed that's the wrong version?  What did I install in the wrong order?

**Comment by realparadyne** [comment]
Adding a global.json to block use of preview SDK's seems to have resolved it.

But this makes it difficult to experiment with new .Net 7 features.
